packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -391,8 +398,12 @@ export async function writeImageToCache(item = {}) {
 function isInvalidCart(error) {
     return !!(
         error.graphQLErrors &&
-        error.graphQLErrors.find(err =>
-            err.message.includes('Could not find a cart')
+        error.graphQLErrors.find(
+            err =>
+                err.message.includes('Could not find a cart') ||
+                err.message.includes(
+                    'The current user cannot perform operations on cart'
```

Comment:
This addition is all that is necessary for the user flow to "work". The other changes in this PR are to make the UI (cart trigger item qty) update properly without having to refresh the page or add another item